<launch>

    <arg name="marker1Id"         default="97"/>
    <arg name="marker2Id"         default="72"/>
    <arg name="marker3Id"         default="185"/>
    <arg name="marker4Id"         default="52"/>

    <arg name="markerSize"        default="0.02"/>    <!-- in m -->
    
    <arg name="eye"               default="left"/>
    
    <arg name="dct_normalization" default="True" />
    <arg name="dct_filter_size"   default="2" />
    
    <arg name="marker1_frame"     default="marker_hand_frame" />
    <arg name="marker2_frame"     default="marker_object_frame" />
    <arg name="marker3_frame"     default="marker_object_frame2" />
    <arg name="marker4_frame"     default="marker_object_frame3" />


    <node pkg="aruco_ros" type="quad" name="aruco_simple" output="screen">    
        
        <remap from="/camera_info" to="/usb_cam/camera_info" />
        <remap from="/image" to="/usb_cam/image_raw" />
        
        <param name="image_is_rectified" value="True"/>
        
        <param name="marker_size" value="$(arg markerSize)"/>
        
        <param name="marker_id1" value="$(arg marker1Id)"/>
        <param name="marker_id2" value="$(arg marker2Id)"/>
        <param name="marker_id3" value="$(arg marker3Id)"/>
        <param name="marker_id4" value="$(arg marker4Id)"/>

        <param name="normalizeImage" value="$(arg dct_normalization)"/>
        <param name="dct_components_to_remove" value="$(arg dct_filter_size)"/>
        
        <param name="parent_name" value="stereo_gazebo_$(arg eye)_camera_optical_frame"/>
        <param name="child_name1" value="$(arg marker1_frame)" />
        <param name="child_name2" value="$(arg marker2_frame)" />
        <param name="child_name3" value="$(arg marker3_frame)" />
        <param name="child_name4" value="$(arg marker4_frame)" />
    </node>

</launch>

